osgar/lib/quaternion.py

```diff
@@ -110,5 +110,58 @@ def angle_between(quaternion0, quaternion1):
     val = 2 * inner_product**2 - 1
     return math.acos(max(-1.0, min(1.0, val)))
 
+def slerp(A, B, t):
+    """
+        Interpolates two quaternions utilizing the Spherical Linear
+            intERPolation (SLERP).
+
+        The input quaternions are given as list of four float values.
+        The order of their coordinates is not important, hence, a quaternion
+            "a + bi + cj + dk" can be passed as, e.g., `[a, b, c, d]`
+            or `[b, c, d, a]`.
+        The coordinates of the result will respect the same order.
+
+        Args:
+            A (list of float): quaternion as a list of four values
+            B (list of float): quaternion as a list of four values
+            t (float): a value between `0.0` and `1.0`
+
+                * if `t == 0.0` then the function returns `A`
+                * if `t == 1.0` then the function returns `B`
+                * if `0.0 < t < 1.0` then the corresponding value on the
+                    shortest path between `A` and `B` is returned
+
+        Returns (list): interpolated quaternion
+    """
+    dotAB = sum(A[i]*B[i] for i in range(4)) # dot product
+    # if the dot product is negative, negate one quaternion to get the shortest
+    # path
+    if dotAB < 0.0:
+        for i in range(4):
+            B[i] = -B[i]
+        dotAB = -dotAB
+    if dotAB > 0.9995:
```

Comment:
here is probably "hidden" assumption that the quaternions are already normalied, right?